### PR TITLE
[Nit] code refactor

### PR DESCRIPTION
**Pull request by SamYuan1990**
It looks like we have some duplicate code, so I just start to make code reusable by ref @KaiyiLiu1234 's previous code change.

btw, what's termination_time meaning? in most case the value is 1 but in this Github action the value is 6.

https://github.com/sustainable-computing-io/kepler-metal-ci/blob/main/.github/workflows/equinix_k8s_flow_churncheck.yml#L83

### CODE REVIEW

File: .github/workflows/equinix_k8s_flow.yml [hunk 1]
```diff
@@ -33,69 +33,14 @@ jobs:
...
-          sudo mv /tmp/k8sconfig /tmp/kubeconfig
-
-      - name: Upload kubeconfig
-        uses: actions/upload-artifact@v4
-        with:
-          name: kubeconfig-${{ github.run_id }}
-          path: /tmp/kubeconfig
-          retention-days: 1
-
+    uses: ./.github/actions/create-k8s-cluster-action
```
Comment (SamYuan1990) [inline .github/workflows/equinix_k8s_flow.yml:36]: ref https://github.com/sustainable-computing-io/kepler-metal-ci/actions/runs/11698335952
@KaiyiLiu1234 , could you please advise how to use self defined github action to make code reusable?

Comment (KaiyiLiu1234) [reply .github/workflows/equinix_k8s_flow.yml:36]: Follow this example: https://github.com/sustainable-computing-io/kepler-metal-ci/blob/main/.github/workflows/equinix_metal_e2e.yml#L39. You need to provide the "name" and use "with" if your action takes inputs. Look at this for an example which uses with: https://github.com/sustainable-computing-io/kepler-metal-ci/blob/main/.github/workflows/equinix_metal_e2e.yml#L67.

Comment (SamYuan1990) [reply .github/workflows/equinix_k8s_flow.yml]: aha, the use here been a part of job define so we need decouple steps from job, and make use as a step with it's own name!

Comment (KaiyiLiu1234) [reply .github/workflows/equinix_k8s_flow.yml]: I think you still need to provide it with a name. setup action should have an example.

Comment (KaiyiLiu1234) [reply .github/workflows/equinix_k8s_flow.yml]: ahh never mind i see the name is added as Create Cluster

File: .github/workflows/equinix_k8s_flow.yml [hunk 2]
```diff
@@ -32,70 +32,14 @@ permissions:
...
+    secrets: inherit
+    with:
+      termination_time: ${{ github.event.inputs.termination_time || '1' }}
+      control_plan: ${{ github.event.inputs.control_plan }}
+      node_plan: ${{ github.event.inputs.node_plan }}
+      metro: ${{ github.event.inputs.metro }}
+      
   Test-K8s:
     name: "Test K8s"
     needs: create-k8s-cluster
```
Comment (KaiyiLiu1234) [inline .github/workflows/equinix_k8s_flow.yml:45]: shouldn't this be Create-k8s-cluster and not create-k8s-cluster?

Comment (SamYuan1990) [reply .github/workflows/equinix_k8s_flow.yml:45]: honestly I don't know but it works. I will have a commit to update it.

Comment (SamYuan1990) [reply .github/workflows/equinix_k8s_flow.yml]: note: I rebased this PR with https://github.com/sustainable-computing-io/kepler-metal-ci/pull/323 and let's try if we need a PR level CI for test our pipeline. ref https://github.com/sustainable-computing-io/kepler-metal-ci/pull/323#issuecomment-2465940106

**Comment by KaiyiLiu1234** [comment]
Going to approve. Make sure to test these workflows to ensure they have expected results (i.e they functionality should be same as before as this is just a code refractor).

**Review by KaiyiLiu1234 — APPROVED** [review]

**Comment by SamYuan1990** [comment]
> Going to approve. Make sure to test these workflows to ensure they have expected results (i.e they functionality should be same as before as this is just a code refractor).

@rootfs , any suggestion for test it before merge? same with [comment](https://github.com/sustainable-computing-io/kepler-metal-ci/pull/323#issuecomment-2465940106)

maybe we need a PR level dummy to keep our core parts been tested. 
@KaiyiLiu1234 , @sthaha wdyt?

**Pull request merged**